### DISCUSSION TITLE
Questions about migrating to v6

### DISCUSSION THREAD

**Discussion by rhegner** [empira/PDFsharp#183] [Q&A] [answered]
For many years I've been using a .NET Core compatible fork of PDFsharp/Migradoc in my ASP.NET Core server application running under Linux. I'm very happy to see that an official .NET Core port is now available.

While transitioning to your new 6.x version I came across two questions:

**Question 1**

In my existing code I have this:

```
var page = document.AddPage();
page.Size = PdfSharp.PageSize.A4;
using (var gfx = XGraphics.FromPdfPage(page))
{
    gfx.MUH = PdfFontEncoding.Unicode;
    docRenderer.RenderPage(gfx, i + 1, PageRenderOptions.All);
}
```

The `MUH` property is no longer available. Is there another way to set font encoding? Or is this no longer needed? (I don't remember why I had to do this, to be honest).

**Question 2**

In my application I had to include the `System.Text.Encoding.CodePages` package and do this initialization somewhere:

```
 // Needed for PDF generation using PDFsharp; otherwise Windows-1252 is missing.
System.Text.Encoding.RegisterProvider(System.Text.CodePagesEncodingProvider.Instance);
```

Is this still needed (under Linux) with v6.x?

**Comment by ThomasHoevel** [comment, marked as answer]
MUH stands for Mega Unicode Hack. The line of code is no longer needed.

PDFsharp and MigraDoc register all encodings they use, so the other line is no longer needed.

**Reply by rhegner** [reply]
Perfect, thanks for the quick response!
